components/board/genericlinux/hw_pwm.go

```diff
@@ -87,6 +88,11 @@ func (pwm *pwmDevice) unexport() error {
 	// device results in an error. We don't care if there's an error: it should be disabled no
 	// matter what.
 	goutils.UncheckedError(pwm.disable())
+
+	// On boards like the Odroid C4, there is a race condition in the kernel where, if you unexport
+	// the pin too quickly after changing something else about it (e.g., disabling it), the whole
+	// PWM system gets corrupted. Sleep for a small amount of time to avoid this.
+	time.Sleep(time.Microsecond)
```

Comment:
This is fine for now, but can we check something's state in the kernel to see if it is unexported in the future? Make a jira ticket.

Nice one, finding where the problem is!

Reply:
If it's already unexported, we return early up on line 80 and skip this. Is that what you meant?